pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -97,18 +101,28 @@ func (cu containersByEvictUnit) NumEvictUnits() int {
 }
 
 // Newest first.
-type byCreated []containerGCInfo
+type byAttempt []containerGCInfo
 
-func (a byCreated) Len() int           { return len(a) }
-func (a byCreated) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
-func (a byCreated) Less(i, j int) bool { return a[i].createTime.After(a[j].createTime) }
+func (a byAttempt) Len() int      { return len(a) }
+func (a byAttempt) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a byAttempt) Less(i, j int) bool {
+	if a[i].attempt == a[j].attempt {
+		return a[i].createTime.After(a[j].createTime)
+	}
+	return a[i].attempt > a[j].attempt
+}
 
 // Newest first.
-type sandboxByCreated []sandboxGCInfo
+type sandboxByAttempt []sandboxGCInfo
 
-func (a sandboxByCreated) Len() int           { return len(a) }
-func (a sandboxByCreated) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
-func (a sandboxByCreated) Less(i, j int) bool { return a[i].createTime.After(a[j].createTime) }
+func (a sandboxByAttempt) Len() int      { return len(a) }
+func (a sandboxByAttempt) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a sandboxByAttempt) Less(i, j int) bool {
+	if a[i].attempt == a[j].attempt {
+		return a[i].createTime.After(a[j].createTime)
```

Comment:
same, it most likely must be opposite condition as this is a `Less` implementation

```suggestion
		return a[i].createTime.Before(a[j].createTime)
```